inst/scripts/generateHGNCHash.R

```diff
@@ -0,0 +1,59 @@
+# generateHGNCHash.R
+
+#' Generates a data frame containing all valid HGNC gene symbols, calls fmatch
+#' on the data frame so it will create a hash of the data frame to be used by
+#' checkGeneSymbols.R.
+#'
+#' @param fURL The URL to the file containing valid HGNC gene symbols.
+#'
+#' @seealso \code{\link{isGeneSymbol}} is used to check if a gene symbol is a
+#' valid HGNC gene symbol that is contained in inst/extdata/HGNCSymbols.rds.
+
+library(fastmatch)
+
+# Open the connection/file.
+## Change fURL to be the URL of the file containing HGNC gene symbols.
+## The file is assumed to contain the symbol, status and locus group of the gene.
+fURL <- "http://tinyurl.com/kk3r9ck"
+HGNCFile <- url(fURL, open = "r")
+
+# Read the header and vertorize.
+header <- readLines(HGNCFile, n = 1)
+headerVector <- unlist(strsplit(header, "\t"))
+
+# Find the index of the columns that contain the gene symbol, status, and locus
+# group of the genes.
+symbolCol <- which(headerVector == "Approved Symbol")
+statusCol <- which(headerVector == "Status")
+locusCol <- which(headerVector == "Locus Group")
+
+# vectors containing the (important) statuses and locus groups of interest.
+impStatuses <- c("Approved")
+impLocusGroups <- c("protein-coding gene", "other", "phenotype")
+
+# Read line by line.
+geneSymbols <- c()
+while (length(entry <- readLines(HGNCFile, n = 1)) > 0) {
```

Comment:
While this will work, (and I'm not requiring you to change this) the "proper" way here would be to use read.delim(), or better readr::read.delim(), select rows to include or exclude, and subset your symbols with a single statement.  No iteration.